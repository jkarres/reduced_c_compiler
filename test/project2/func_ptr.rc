
function : void f() { cout << "in f()" << endl; }

function : void g() { cout << "in g()" << endl; }

function : void main() {


  // initialization from function name          
  funcptr : void () fp1 = f;
  fp1();

  // assignment from function name
  fp1 = g;
  fp1();

  // initialization from function pointer
  fp1 = f;
  funcptr : void () fp2 = fp1;
  fp2();

  // assignment from function pointer
  fp1 = g;
  fp2 = fp1;
  fp2();

}
