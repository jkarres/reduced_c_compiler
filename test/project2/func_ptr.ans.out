in f()
in g()
in f()
in g()
